File: app/src/main/java/net/squanchy/injection/ActivityContextModule.java
package net.squanchy.injection;

import android.app.Activity;
import android.support.v7.app.AppCompatActivity;

import dagger.Module;
import dagger.Provides;

@Module
public class ActivityContextModule {

    private final AppCompatActivity activity;

    public ActivityContextModule(AppCompatActivity activity) {
        this.activity = activity;
    }

    @Provides
    AppCompatActivity appCompatActivityContext() {
        return activity;
    }

    @Provides
    Activity activityContext() {
        return activity;
    }
}

File: app/src/main/java/net/squanchy/injection/ActivityContextModule.kt
package net.squanchy.injection

import android.app.Activity
import androidx.appcompat.app.AppCompatActivity

import dagger.Module
import dagger.Provides

@Module
class ActivityContextModule(private val activity: AppCompatActivity) {

    @Provides
    internal fun appCompatActivityContext(): AppCompatActivity = activity

    @Provides
    internal fun activityContext(): Activity = activity
}
